src/battle_util.c

```diff
@@ -3871,6 +3872,10 @@ u8 AtkCanceller_UnableToUseMove(void)
                 {
                     SetRandomMultiHitCounter();
                 }
+                
+                if (gMultiHitCounter < 4 && GetBattlerHoldEffect(gBattlerAttacker, TRUE) == HOLD_EFFECT_LOADED_DICE)
+                    gMultiHitCounter = 4;
```

Comment:
Loaded dice doesn't set the counter to 4, it's meant to be *at least* 4, so this should go in ``SetRandomMultiHitCounter`.

Here's the detailed info from the Smogon forums: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-9#post-9400247